### PR TITLE
[KNOX-185]: Creation of gha-k8s-namespace GitHub Action

### PR DESCRIPTION
**Pull request by alehechka**
The purpose of this GitHub Action is to automate the creation of ExternalName services to allow for communication with other applications deployed into the main namespace without having to deploy a new copy into the feature branch namespace.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,57 @@
...
+    uses: dmsi-io/gha-k8s-namespace@v1
+    with:
+      GCP_SA_KEY: ${{ secrets.GCP_SA_KEY }}
+      GKE_CLUSTER_NAME: ${{ secrets.GCP_STAGING_CLUSTER_NAME }}
+      GCP_ZONE: ${{ secrets.GCP_ZONE }}
+      GCP_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+      TLD: ${{ secrets.TOP_LEVEL_DOMAIN }}
+```
+
+> It is recommended to leave the if statement on this step to only run this action on creation of new branches. However, this action is not destructive in nature an could be run on every commit.
```
Comment (nathanHotz) [inline README.md:20]: typo, and*

**Comment by alehechka** [comment]
GHA in action: https://github.com/dmsi-io/door-maint-api/blob/feature/KNOX-185/.github/workflows/main.yaml

Example run of namespace creation: https://github.com/dmsi-io/door-maint-api/actions/runs/1560716959

Example run of skipping namespace creation: https://github.com/dmsi-io/door-maint-api/runs/4476329172

**Review by nathanHotz — APPROVED** [review]

File: action.yaml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        done
+      shell: bash
+
+    - name: Deploy Default Deployments
+      run: |
+        REPOS="${{ inputs.repos }}"
+        for repo in ${REPOS//,/ }
+        do
+            if [[ $(kubectl get deployment $repo -n $TO_NAMESPACE -o=name --ignore-not-found) ]]; then
+                echo "$repo deployment already exists in $TO_NAMESPACE"
```
Comment (brianspowers) [inline action.yaml:123]: awesome

**Review by brianspowers — APPROVED** [review]
This is great, I'm really excited to have this in action.

**Pull request merged**